### ISSUE TITLE
ExtentCucumberFormatter is not compatible with cucumber-jvm 2.0.0

### ISSUE THREAD

**Issue by mpkorstanje** [email2vimalraj/CucumberExtentReporter#52]
The upcoming release of `cucumber-jvm:2.0.0` will use an event based system rather then the hook based system. This means that [ExtentCucumberFormatter](https://github.com/email2vimalraj/CucumberExtentReporter/blob/master/src/main/java/com/cucumber/listener/ExtentCucumberFormatter.java) will not work with cucumber 2.0.0.

To fix this the ExtentCucumberFormatter should implement [cucumber.api.formatter.Formatter](https://github.com/cucumber/cucumber-jvm/blob/master/core/src/main/java/cucumber/api/formatter/Formatter.java) and then listen to [these events](https://github.com/cucumber/cucumber-jvm/tree/master/core/src/main/java/cucumber/api/event). For a concrete example please see the [JSONFormatter](https://github.com/cucumber/cucumber-jvm/blob/master/core/src/main/java/cucumber/runtime/formatter/JSONFormatter.java).

**Comment by email2vimalraj** [comment]
@mpkorstanje : Thanks for noticing me. I'll look into this and update accordingly.

**Comment by krmahadevan** [comment]
@mpkorstanje - Any place wherein I can look up on what is coming in 2.0 ? I would be particularly interested in knowing if the 2.0 is going to come out with parallelism support (especially at the scenario levels).

**Comment by mpkorstanje** [comment]
Sure: https://github.com/cucumber/cucumber-jvm/blob/master/History.md

Or try:

```
<repository>
    <id>sonatype-snapshots</id>
    <url>https://oss.sonatype.org/content/repositories/snapshots</url>
    <snapshots>
        <enabled>true</enabled>
    </snapshots>
</repository>

<dependency>
    <groupId>io.cucumber</groupId>
    <artifactId>cucumber-java</artifactId>
    <version>2.0.0-SNAPSHOT</version>
    <scope>test</scope>
</dependency>
```

But it is mostly bug fixes. Would recommend looking at:

https://github.com/temyers/cucumber-jvm-parallel-plugin

And if you fix https://github.com/temyers/cucumber-jvm-parallel-plugin/issues/127 you'll have good support for extents in parallel mode.

**Comment by MaximeRyckmans** [comment]
Any update on this? I think it would be a very nice enhancement, as cucumber 2.0 is live now.
No pressure of course, you have already done so much ;).

**Comment by email2vimalraj** [comment]
@MaximeRyckmans working on it.

**Comment by krmahadevan** [comment]
@MaximeRyckmans - Would you know if Cucumber 2.0 provides thread safe reports ? I am working on something that needs thread safe reports.

**Comment by MaximeRyckmans** [comment]
@krmahadevan I don't think so, have a look at this link: https://github.com/cucumber/cucumber-jvm/issues/630
